src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -245,25 +250,26 @@ public String getDisplayName() {
 
 
         public FormValidation doValidateForm(
-                @QueryParameter("restApiKey") final String restApiKey,
+                @QueryParameter("restApiKey") final Secret restApiKey,
                 @QueryParameter("environmentId") final String environmentId,
                 @QueryParameter("applicationId") final String applicationId
         ) {
             return MablStepBuilderValidator.validateForm(restApiKey, environmentId, applicationId);
         }
 
-        public ListBoxModel doFillApplicationIdItems(@QueryParameter String restApiKey, @QueryParameter boolean disableSslVerification) {
-            if(restApiKey == null || restApiKey.isEmpty()) {
+        public ListBoxModel doFillApplicationIdItems(@QueryParameter Secret restApiKey, @QueryParameter boolean disableSslVerification) {
+
+            if(restApiKey == null || restApiKey.getPlainText() == null || restApiKey.getPlainText().isEmpty()) {
```

Comment:
Seems like logic is being repeated 3x. Could move to a static helper. e.g. `boolean isSecretEmpty(Secret value)` for easier testing.